drivers/input/input_vs1838b.c

```diff
@@ -0,0 +1,361 @@
+
+/* Empiric tolerance values. Might be a good idea to put them in the Kconfig? */
+#define VS1838B_NEC_LEAD_PULSE_PERIOD_TOLERANCE_USEC 400
+#define VS1838B_NEC_BIT_DETECT_PERIOD_TOLERANCE_USEC 150
+#define VS1838B_NEC_BIT_0_TOTAL_TOLERANCE_USEC       200
+#define VS1838B_NEC_BIT_1_TOTAL_TOLERANCE_USEC       200
+
+/* Tick ranges for the NEC elements */
+#define VS1838B_NEC_LEAD_PULSE_ON_MIN_TICK                                                         \
+	VS1838B_MIN_TICK(NEC_LEAD_PULSE_PERIOD_ON_USEC,                                            \
+			 VS1838B_NEC_LEAD_PULSE_PERIOD_TOLERANCE_USEC)
+#define VS1838B_NEC_LEAD_PULSE_ON_MAX_TICK                                                         \
+	VS1838B_MAX_TICK(NEC_LEAD_PULSE_PERIOD_ON_USEC,                                            \
+			 VS1838B_NEC_LEAD_PULSE_PERIOD_TOLERANCE_USEC)
+
+#define VS1838B_NEC_LEAD_PULSE_OFF_MIN_TICK                                                        \
+	VS1838B_MIN_TICK(NEC_LEAD_PULSE_PERIOD_OFF_USEC,                                           \
+			 VS1838B_NEC_LEAD_PULSE_PERIOD_TOLERANCE_USEC)
+#define VS1838B_NEC_LEAD_PULSE_OFF_MAX_TICK                                                        \
+	VS1838B_MAX_TICK(NEC_LEAD_PULSE_PERIOD_OFF_USEC,                                           \
+			 VS1838B_NEC_LEAD_PULSE_PERIOD_TOLERANCE_USEC)
+
+#define VS1838B_NEC_BIT_DETECT_MIN_TICK                                                            \
+	VS1838B_MIN_TICK(NEC_BIT_DETECT_PERIOD_USEC, VS1838B_NEC_BIT_DETECT_PERIOD_TOLERANCE_USEC)
+#define VS1838B_NEC_BIT_DETECT_MAX_TICK                                                            \
+	VS1838B_MAX_TICK(NEC_BIT_DETECT_PERIOD_USEC, VS1838B_NEC_BIT_DETECT_PERIOD_TOLERANCE_USEC)
+
+#define VS1838B_NEC_BIT_0_TOTAL_MIN_TICK                                                           \
+	VS1838B_MIN_TICK(NEC_BIT_0_TOTAL_PERIOD_USEC, VS1838B_NEC_BIT_0_TOTAL_TOLERANCE_USEC)
+#define VS1838B_NEC_BIT_0_TOTAL_MAX_TICK                                                           \
+	VS1838B_MAX_TICK(NEC_BIT_0_TOTAL_PERIOD_USEC, VS1838B_NEC_BIT_0_TOTAL_TOLERANCE_USEC)
+
+#define VS1838B_NEC_BIT_1_TOTAL_MIN_TICK                                                           \
+	VS1838B_MIN_TICK(NEC_BIT_1_TOTAL_PERIOD_USEC, VS1838B_NEC_BIT_1_TOTAL_TOLERANCE_USEC)
+#define VS1838B_NEC_BIT_1_TOTAL_MAX_TICK                                                           \
+	VS1838B_MAX_TICK(NEC_BIT_1_TOTAL_PERIOD_USEC, VS1838B_NEC_BIT_1_TOTAL_TOLERANCE_USEC)
+
+struct vs1838b_data {
+	struct device const *dev;
+	struct gpio_callback input_cb;
+	struct k_work_delayable decode_work;
+	int64_t edges_ticks[NEC_SINGLE_COMMAND_EDGES_COUNT];
+	uint8_t edges_count;
+	struct k_sem decode_sem;
+};
+
+struct vs1838b_config {
+	struct gpio_dt_spec input;
+};
+
+/******************************************************************************
+ * Internal stuff
+ *******************************************************************************/
```

Comment:
it's a small driver no need to put fancy decorations and sections, but if you really really want to keep it simple and consistent, just

```
/*
 * Internal stuff
 */
```

would be loud enough.

I would just drop them all